src/ios-deploy/ios-deploy.m

```diff
@@ -1037,15 +1065,15 @@ void start_remote_debug_server(AMDeviceRef device) {
             default:
                 check_error(start_err);
         }
-        check_error(AMDeviceSecureStartService(device, CFSTR("com.apple.debugserver"), NULL, &con));
+        check_error(AMDeviceSecureStartService(device, serviceName, NULL, &dbgServiceConnection));
     }
-    assert(con != NULL);
-    gdbfd = AMDServiceConnectionGetSocket(con);
+    assert(dbgServiceConnection != NULL);
+
     /*
      * The debugserver connection is through a fd handle, while lldb requires a host/port to connect, so create an intermediate
      * socket to transfer data.
      */
-    server_socket = CFSocketCreateWithNative (NULL, gdbfd, kCFSocketDataCallBack, &server_callback, NULL);
+    server_socket = CFSocketCreateWithNative (NULL, AMDServiceConnectionGetSocket(dbgServiceConnection), kCFSocketDataCallBack, &server_callback, NULL);
```

Comment:
You might need to change to this to subscribe to kCFSocketReadCallBack. kCFSocketDataCallBack will consume the incoming data so AMDServiceConnectionReceive will hang because no data is left. kCFSocketReadCallBack will be called when data is available to read but won't read it.

Reply:
I just tried this patch with `kCFSocketDataCallBack` and the debugger successfully connected.

Reply:
That's interesting, on my setup this patch does not work for the reason mentioned above.

Do you mind sharing your setup? I have:

- iPad Air 2 running iOS14 beta 6 (18A5357e)
- iMac Pro running macOS 10.15.6 (19G73)
- Xcode 12.0 beta 6 (12A8189n)

Reply:
Yeah, it should be kCFSocketReadCallBack.  Got lost while fixing merge conflicts. I'll fix the pull request tomorrow.